### PR TITLE
[bitnami/solr] Update readinessProbe and livenessProbe

### PR DESCRIPTION
**Pull request by gongomgra**
Signed-off-by: Gonzalo Gomez Gracia <gonzalog@vmware.com>

**Description of the change**

Updates the `livenessProbe` and `readinessProbe` definitions to execute internal `solr` binary commands that check the status of the cluster and the Solr service itself. Also recovers `livenessProbe` and `readinessProbe` settings for `initialDelaySeconds`, `periodSeconds`, `timeoutSeconds`, `failureThreshold` and `successThreshold` in the StatefulSet template. 

Additionally, this PR removes the `goss-wait.yaml` file that is no longer needed.

**Benefits**

More confident `livenessProbe` and `readinessProbe` actions.

**Possible drawbacks**

Not expected.

**Checklist**

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the values.yaml and added to the README.md using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/solr/Chart.yaml [hunk 1]
```diff
@@ -27,4 +27,4 @@ name: solr
 sources:
   - https://github.com/bitnami/bitnami-docker-solr
   - https://lucene.apache.org/solr/
-version: 4.0.5
+version: 4.1.0
```
Comment (miguelaeh) [inline bitnami/solr/Chart.yaml:30]: I would say this is a patch more than a minor version

**Review by miguelaeh — APPROVED** [review]
LGTM! But check the version comment

**Review by joancafom — DISMISSED** [review]
LGTM

**Review by miguelaeh — APPROVED** [review]
LGTM!

**Review by joancafom — DISMISSED** [review]
LGTM!

**Pull request merged**